examples/fidelity.py

```diff
@@ -0,0 +1,86 @@
+"""Simulates the fidelity estimation.
```

Comment:
"fidelity estimation" is a very generic thing one might do. Conventionally it's done by tomography

Specify "direct fidelity estimation", which is a specific thing given by the referenced papers. The filename should also be more specific